modules/web-console/web-console-common/src/main/java/org/apache/ignite/console/event/Event.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.ignite.console.event;
+
+import org.springframework.context.ApplicationEvent;
+
+/**
+ * Generic application event with payload
+ */
+public class Event<T> extends ApplicationEvent {
+    /** Type. */
+    private Type type;
+
+    /** Payload. */
+    private T payload;
+
+    /**
+     * @param type Type.
+     * @param payload Payload.
+     */
+    public Event(Type type, T payload) {
+        super(new Object());
```

Comment:
Why not payload? Look at JaasAuthenticationEvent

Reply:
Done.